pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -823,6 +823,12 @@ func (s *DiscoveryServer) AdsPushAll(version string, push *model.PushContext,
 		version, len(push.Services(nil)), adsClientCount())
 	monServices.Set(float64(len(push.Services(nil))))
 
+	onlyMesh := make(map[string]bool)
+	onlyMesh[model.IstioMeshGateway] = true
+	numberOfKnownMeshVirtualServices := len(push.VirtualServices(nil, onlyMesh))
+	adsLog.Debugf("Known virtual services for mesh gateway:%d", numberOfKnownMeshVirtualServices)
+	monVServices.Set(float64(numberOfKnownMeshVirtualServices))
```

Comment:
I think this is still a bit misleading to scope it just to the gateways. Some people don't even use gateways for example. Maybe we can just set it in `initVirtualServices`? At that point we definitely have the correct number of virtual services

Reply:
That sounds reasonable, I will have a look. Thanks for the hint.